controller/db/router_store_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openziti/foundation/storage/boltz"
+	"go.etcd.io/bbolt"
+)
+
+func Test_RouterStore(t *testing.T) {
+	ctx := NewTestContext(t)
+	defer ctx.Cleanup()
+	ctx.Init()
+
+	t.Run("test create invalid api routers", ctx.testCreateInvalidRouters)
+	t.Run("test create router", ctx.testCreateRouters)
+	t.Run("test load/query routers", ctx.testLoadQueryRouters)
+	t.Run("test update routers", ctx.testUpdateRouters)
+	t.Run("test delete routers", ctx.testDeleteRouters)
+}
+
+func (ctx *TestContext) testCreateInvalidRouters(_ *testing.T) {
+	defer ctx.cleanupAll()
```

Comment:
If these tests fail and you don't use the `T` passed in here they will exit with weird errors as `ctx.Req` will call fail now outside of its own context. It is undefined behavior.

Reply:
Fixed